core/src/main/java/org/infinispan/marshall/core/GlobalMarshaller.java

```diff
@@ -33,6 +35,8 @@
 import org.infinispan.marshall.core.ClassToExternalizerMap.IdToExternalizerMap;
 import org.infinispan.util.logging.Log;
 import org.infinispan.util.logging.LogFactory;
+import org.jboss.marshalling.ObjectTable;
```

Comment:
This is not right the right way. GlobalMarshaller is independent from underlying marshalling library. It should not be depending on JBoss Marshalling itself. Any JBoss Marshalling code should be limited to ExternalJBossMarshaller.

Reply:
Ok, I can change this. FYI the GlobalMarshaller as it currently is implemented will be refactored heavily with https://issues.jboss.org/browse/ISPN-9622, so ^ code should be temporary.

Reply:
Fair enough. Just make sure that when you refactor, 3rd party marshalling code is separated away.